### PR TITLE
api spec editorial fixes

### PR DESCRIPTION
**Pull request by bashbaug**
This PR contains a number of formatting and grammatical fixes I found in the API spec.  There are no functional changes.

### CODE REVIEW

File: api/opencl_runtime_layer.asciidoc [hunk 1]
```diff
@@ -8565,7 +8562,7 @@ include::{generated}/api/version-notes/clGetEventInfo.asciidoc[]
 
 [[event-info-table]]
 .List of supported param_names by <<clGetEventInfo>>
-[width="100%",cols="<30%,<33%,<37%",options="header"]
+[width="100%",cols="<33%,<17%,<50%",options="header"]
 |====
 | Event Info | Return Type | Info. returned in _param_value_
```
Comment (EwanC) [inline api/opencl_runtime_layer.asciidoc:8567]: This is very minor, but I noticed it yesterday in other `clGetFooInfo` entry-points. Should the period after "Info" exist in the third column heading? Think it reads better without it. Feel free not to make this change now even if so to make any spec release time constraints

Comment (bashbaug) [reply api/opencl_runtime_layer.asciidoc]: Thanks, good suggestion.  I've changed this to use "Description" as the column header consistently, which was already a convention used by other table cells.  If we want to use a more descriptive column header in the future this is easy enough to change, but for now I think the consistency is a nice improvement.  Thanks again!

**Review by alycm — APPROVED** [review]
LGTM, two tiny comments that could be ignored.

File: api/opencl_platform_layer.asciidoc [hunk 2]
```diff
@@ -1719,7 +1720,8 @@ unit, pass the following in properties argument:
 [source,c]
 ----
 { CL_DEVICE_PARTITION_BY_COUNTS,
-  3, 1, CL_DEVICE_PARTITION_BY_COUNTS_LIST_END, 0 }
+    3, 1, CL_DEVICE_PARTITION_BY_COUNTS_LIST_END,
```
[the remark this answers is not in the document]
Comment (bashbaug) [reply api/opencl_platform_layer.asciidoc:1723]: Let's leave this as-is for now.

File: api/opencl_runtime_layer.asciidoc [hunk 3]
```diff
@@ -6573,9 +6572,9 @@ include::{generated}/api/version-notes/clGetProgramInfo.asciidoc[]
 
 [[program-info-table]]
 .List of supported param_names by <<clGetProgramInfo>>
-[width="100%",cols="<34%,<33%,<33%",options="header"]
+[width="100%",cols="<33%,<17%,<50%",options="header"]
```
[the remark this answers is not in the document]
Comment (bashbaug) [reply api/opencl_runtime_layer.asciidoc:6575]: I've fixed this by removing the spaces between the type and `*` in all of the tables.

**Pull request merged**